File: examples/java/minimal-order-service/src/main/java/com/example/order/application/ports/outbound/PaymentGateway.java
package com.example.order.application.ports.outbound;

import com.example.order.domain.Money;

/**
 * Outbound port: defines what the application needs from payment processing.
 * Implemented by adapters (real payment service, fake for testing, etc.)
 */
public interface PaymentGateway {
    PaymentResult processPayment(Money amount, String orderId);
}

File: examples/go/minimal-order-service/application/ports/outbound/payment_gateway.go
package outbound

import "github.com/example/order-service/domain"

// PaymentResult represents the result of a payment operation.
type PaymentResult struct {
	Success       bool
	TransactionID string
	ErrorMessage  string
}

// PaymentGateway is the outbound port for payment processing.
type PaymentGateway interface {
	ProcessPayment(amount *domain.Money, orderID string) (*PaymentResult, error)
}
